pkg/component/operator/web/v0/crawl_website.go

```diff
@@ -177,12 +242,41 @@ func (e *execution) CrawlWebsite(input *structpb.Struct) (*structpb.Struct, erro
 		inputStruct.URL = "https://" + inputStruct.URL
 	}
 
+	scrapeDone := make(chan struct{})
 	go func() {
+		defer close(scrapeDone)
 		_ = c.Visit(inputStruct.URL)
 		c.Wait()
 	}()
 
-	<-ctx.Done()
+	// To avoid to wait for 2 minutes, we use a timer to check if there is a new page.
+	// If there is no new page, we cancel the context.
+	inactivityTimer := time.NewTimer(2 * time.Second)
+	defer inactivityTimer.Stop()
+
+	// There are 4 cases to finish the program:
+	// 1. No more pages to scrape: c.Wait() returns and the goroutine closes scrapeDone. Program finishes before the timeout.
+	// 2. Max pages scraped: c.OnResponse cancels the context / closes scrapeDone. Program finishes before the timeout.
+	// 3. Max pages haven't been collected before the timeout. Context is canceled and program finishes at the timeout.
+	// 4. Max pages haven't been collected before the timeout. Context is canceled and the program finishes when there are no more validate data in 2 seconds.
```

Comment:
```suggestion
	// 4. Max pages haven't been collected but the delay between completed requests has increased over 2 seconds. We consider this data invalid because (insert reason here), so we avoid waiting for the timeout.
```